datatypes/src/raster/tiling.rs

```diff
@@ -0,0 +1,206 @@
+    }
+
+    /// generates the tile idx for the tiles intersecting the bounding box
+    pub fn tile_information_iterator(
+        &self,
+        bounding_box: BoundingBox2D,
+    ) -> impl Iterator<Item = TileInformation> {
+        let tile_pixel_size = self.tile_pixel_size;
+        let geo_transform = self.geo_transform;
+        self.tile_idx_iterator(bounding_box)
+            .map(move |idx| TileInformation::new(idx, tile_pixel_size, geo_transform))
+    }
+}
+
+/// The `TileInformation` is used to represent the spatial position of each tile
+#[derive(PartialEq, Debug, Copy, Clone, Serialize, Deserialize)]
+pub struct TileInformation {
+    pub tile_size_in_pixels: GridShape2D,
+    pub global_tile_position: GridIdx2D,
+    pub global_geo_transform: GeoTransform,
+}
+
+impl TileInformation {
+    pub fn new(
+        global_tile_position: GridIdx2D,
+        tile_size_in_pixels: GridShape2D,
+        global_geo_transform: GeoTransform,
+    ) -> Self {
+        Self {
+            global_tile_position,
+            tile_size_in_pixels,
+            global_geo_transform,
+        }
+    }
+
+    #[allow(clippy::unused_self)]
+    pub fn local_upper_left_idx(&self) -> GridIdx2D {
+        [0, 0].into()
+    }
+
+    pub fn local_lower_left_idx(&self) -> GridIdx2D {
+        [self.tile_size_in_pixels.axis_size_y() as isize - 1, 0].into()
+    }
+
+    pub fn local_upper_right_idx(&self) -> GridIdx2D {
+        [0, self.tile_size_in_pixels.axis_size_x() as isize - 1].into()
+    }
+
+    pub fn local_lower_right_idx(&self) -> GridIdx2D {
+        let GridIdx([y, _]) = self.local_lower_left_idx();
+        let GridIdx([_, x]) = self.local_upper_right_idx();
+        [y, x].into()
+    }
+
+    pub fn global_tile_position(&self) -> GridIdx2D {
+        self.global_tile_position
+    }
+
+    pub fn global_upper_left_idx(&self) -> GridIdx2D {
+        let [tile_size_y, tile_size_x] = self.tile_size_in_pixels.into_inner();
+        self.global_tile_position() * [tile_size_y as isize, tile_size_x as isize]
+    }
+
+    pub fn global_upper_right_idx(&self) -> GridIdx2D {
+        self.global_upper_left_idx() + self.local_upper_right_idx()
+    }
+
+    pub fn global_lower_right_idx(&self) -> GridIdx2D {
+        self.global_upper_left_idx() + self.local_lower_right_idx()
+    }
+
+    pub fn global_lower_left_idx(&self) -> GridIdx2D {
+        self.global_upper_left_idx() + self.local_lower_left_idx()
+    }
+
+    pub fn tile_size_in_pixels(&self) -> GridShape2D {
+        self.tile_size_in_pixels
+    }
+
+    pub fn local_to_global_idx(&self, local_pixel_position: GridIdx2D) -> GridIdx2D {
+        self.global_upper_left_idx() + local_pixel_position
+    }
+
+    pub fn tile_geo_transform(&self) -> GeoTransform {
+        let tile_upper_left_coord = self
+            .global_geo_transform
+            .grid_idx_to_coordinate_2d(self.global_upper_left_idx());
```

Comment:
don't we use use the center for the geo_transform reference?

Reply:
well that depends how you look at it. The GeoTransform does not know about a center. all it does is take a coordinate and a pixel size. In a grid it binds the pixel at pixel coordinate 0,0 to a coordinate. By convention this is the upper left coordinate. If you look at GDAL it will always bind the upper left pixel of a raster (which is 0,0 in GDAL) to a coordinate.